synapse/handlers/e2e_keys.py

```diff
@@ -1045,8 +1032,11 @@ def _get_e2e_cross_signing_verify_key(self, user_id, key_type, from_user_id=None
             logger.debug("No %s key found for %s", key_type, user_id)
             raise NotFoundError("No %s key found for %s" % (key_type, user_id))
 
-        key_id, verify_key = get_verify_key_from_cross_signing_key(key)
-        return key, key_id, verify_key
+        try:
+            key_id, verify_key = get_verify_key_from_cross_signing_key(key)
+            return key, key_id, verify_key
+        except ValueError:
+            raise SynapseError(502, "Invalid %s key retrieved from remote server")
```

Comment:
* not necessarily retrieved from a remote server
* %s needs a value
* we should avoid throwing away the ValueError without logging it
* the `try/except` should just wrap the line we think might throw.